vortex-scalar/src/primitive.rs

```diff
@@ -300,17 +300,51 @@ impl From<usize> for Scalar {
 }
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+/// Binary element-wise operations on two arrays or two scalars.
 pub enum BinaryNumericOperator {
+    /// Binary element-wise addition of two arrays or of two scalars.
     Add,
+    /// Binary element-wise subtraction of two arrays or of two scalars.
     Sub,
+    /// Same as [BinaryNumericOperator::Sub] but with the parameters flipped: `right - left`.
+    RSub,
+    /// Binary element-wise multiplication of two arrays or of two scalars.
     Mul,
+    /// Binary element-wise division of two arrays or of two scalars.
     Div,
+    /// Same as [BinaryNumericOperator::Div] but with the parameters flipped: `right - left`.
+    RDiv,
     // Missing from arrow-rs:
     // Min,
     // Max,
     // Pow,
 }
 
+impl BinaryNumericOperator {
+    pub fn swap(self) -> Self {
+        match self {
+            BinaryNumericOperator::Add => BinaryNumericOperator::Add,
+            BinaryNumericOperator::Sub => BinaryNumericOperator::RSub,
+            BinaryNumericOperator::RSub => BinaryNumericOperator::Sub,
+            BinaryNumericOperator::Mul => BinaryNumericOperator::Mul,
+            BinaryNumericOperator::Div => BinaryNumericOperator::RDiv,
+            BinaryNumericOperator::RDiv => BinaryNumericOperator::Div,
+        }
+    }
+
+    pub fn math_symbol(&self) -> String {
```

Comment:
What is this used for? If it's for display, it should just be the `Display` trait. But I don't think we should do this, since  it feels like a foot-gun when printing, e.g. `format!("{lhs} {op} {rhs}", ...)` would be wrong for RDiv and Sub.

Reply:
Agreed. Too clever. Nixed.